apps/admin-server/src/pages/projects/[project]/settings/resource/index.tsx

```diff
@@ -59,6 +60,11 @@ type Tag = {
   type: string;
 };
 
+type Status = {
+  id: number;
```

Comment:
Kunnen we id velden strings maken? We gaan niet rekenen of parsen met ids en mocht er ooit de wens komen om uuids of iets anders te gebruiken voor id dan hoeven we niets aan te passen.

Daarnaast wellicht handig om types van models in de types package in packages te stoppen? Dan kunnen we die hergebruiken op eventueel andere plekken

Reply:
In de regel er boven staat voor tags precies hetzelfde :)

Als je er strings van maakt dan kun  je (dus) ids als strings naar de api sturen die vervolgens gaat stuiteren omdat die integers verwacht. Het hele punt van typen lijkt me dat je dat soort dingen wilt voorkomen (compile time herkennen van fouten ipv pas runtime). Het argument 'misschien maken we er ooit nog strings van' lijkt me dan minder steekhoudend.

Wat betreft die missende velden: dit is alleen de koppling naar default waarden op een project; de rest heb je niet nodig.

Reply:
Mijn vraag is eerder, waarom worden ids ook als integers verwacht in de api? Je gaat niet rekenen met de ids en de parsing slag is dan een onnodige overhead naar mijn idee.

Reply:
@LorenzoJokhan In de database is het `id` van een resource een INT. Imo wil je de TypeScript types zo dicht mogelijk tegen je daadwerkelijke data aan laten zitten, in dit geval dus een `number`.

Een migratie naar UUIDs wordt dan ook niet enkel door het type tegengehouden, maar ook door de databasestructuur.

Reply:
@rudivanhierden ben het eens met die statement. Dat is dan wel naar mijn mening een flawed database design en zou die column met een migratie omgezet moeten worden naar een varchar (ofzoiets) maar wellicht iets om over te denken voor ooit in de technical debt. Kan denk ik idd geen kwaad om het voor nu zo te laten